python/lsst/meas/extensions/piff/piffPsfDeterminer.py

```diff
@@ -585,12 +588,15 @@ def _get_threshold(nth_order):
 
         drawSize = 2*np.floor(0.5*stampSize/self.config.samplingSize) + 1
 
-        used_image_pos = [s.image_pos for s in piffResult.stars if not s.is_flagged and not s.is_reserve]
+        used_image_starId = {s.data.properties['starId'] for s in piffResult.stars
+                             if not s.is_flagged and not s.is_reserve}
```

Comment:
Making this a `set` is good. Order doesn't matter and membership check is a tad bit faster.